src/status_im/ui/screens/profile/events.cljs

```diff
@@ -39,43 +40,129 @@
       {:dispatch [:navigation-replace :chat identity]})))
 
 (handlers/register-handler-fx
-  :my-profile/edit
-  (fn [{:keys [db]} [_ edit-type edit-value]]
-    (let [current-account-id (:accounts/current-account-id db)
-          current-account (-> db
-                              (get-in [:accounts/accounts current-account-id])
-                              (select-keys account-profile-keys))
-          new-db (-> db
-                     (update-in [:my-profile/edit] merge current-account)
-                     (assoc-in [:my-profile/edit :edit-status?] (= edit-type :status true)))]
-      {:db new-db
-       :dispatch [:navigate-to :edit-my-profile]})))
-
-(handlers/register-handler-fx
-  :my-profile/update-status
-  (fn [_ [_ new-status]]
-    (when-not (string/blank? new-status)
-      {:dispatch-n [[:check-status-change new-status]
-                    [:account-update {:status new-status}]]})))
-
-(handlers/register-handler-fx
   :my-profile/update-phone-number
   ;; Switch user to the console issuing the !phone command automatically to let him change his phone number.
   ;; We allow to change phone number only from console because this requires entering SMS verification code.
   (fn [_ _]
     {:dispatch-n [[:navigate-to :chat console-chat-id]
                   [:select-chat-input-command {:name "phone"}]]}))
 
+(defn get-current-account [{:keys [:accounts/current-account-id] :as db}]
+  (get-in db [:accounts/accounts current-account-id]))
+
+(handlers/register-handler-fx
+  :my-profile.drawer/edit-name
+  (fn [{:keys [db]} _]
+    (let [{:my-profile/keys [default-name edit auto-save]} db
+          {:keys [name public-key]}                        (get-current-account db)]
+      {:db (cond-> db
+             (not default-name) (assoc :my-profile/default-name (gfycat/generate-gfy public-key))
+             :always            (assoc-in [:my-profile/drawer :name] name))})))
```

Comment:
I am more used to see `:else`. I don't know if we have such convention in the project.

Reply:
We used `true` in some places, but i like `:always` much more than `true` tbh. And `:else` isn't quite correct here, because other branches can be executed as well.

Reply:
Indeed `:always` is more appropriate for `cond->`.